src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -213,7 +214,22 @@ export class DataSourceSavedObjectsClientWrapper {
 
           return {
             ...attributes,
-            auth: await this.encryptCredentials(auth, encryptionContext),
+            auth: await this.encryptBasicAuthCredential(auth, encryptionContext),
+          };
+        } else {
+          return attributes;
+        }
+      case AuthType.SigV4:
+        // Fetch and validate existing signature
+        if (credentials?.accessKey && credentials?.secretKey) {
```

Comment:
if the request include only one of accessKey or secretKey, we should throw exception

Reply:
good point